genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/repositories/JpaCommandRepository.java

```diff
@@ -47,6 +47,23 @@
             + " AND created >= :jobCreatedThreshold"
             + ")";
 
+    /**
+     * The query to find the id's of commands that should be marked inactive due to not being used.
+     */
+    // NOTE TO FUTURE SELF: JPQL does not support limit and wants you to use page size etc. Not worth hassle.
+    String FIND_OLD_COMMANDS_QUERY =
+        "SELECT id"
+            + " FROM commands"
+            + " WHERE status IN (:currentStatuses)"
+            + " AND created < :commandCreatedThreshold"
+            + " AND id NOT IN ("
+            + "SELECT DISTINCT(command_id)"
+            + " FROM jobs"
+            + " WHERE command_id IS NOT NULL"
+            + " AND created >= :jobCreatedThreshold"
+            + ")"
+            + " LIMIT :limit";
```

Comment:
I was thinking whether we should `ORDER BY` but that'd probably be unnecessary. The commands will be marked as `INACTIVE` and will be excluded in the next batch.